plugins/hls-change-type-signature-plugin/src/Ide/Plugin/ChangeTypeSignature.hs

```diff
@@ -0,0 +1,247 @@
+                                                 handleMaybeM, response)
+import           Ide.Types                      (PluginDescriptor (..),
+                                                 PluginId, PluginMethodHandler,
+                                                 defaultPluginDescriptor,
+                                                 mkPluginHandler)
+import           Language.LSP.Types             (CodeAction (..),
+                                                 CodeActionContext (CodeActionContext),
+                                                 CodeActionKind (CodeActionQuickFix),
+                                                 CodeActionParams (..), Command,
+                                                 Diagnostic (..), List (..),
+                                                 Method (TextDocumentCodeAction),
+                                                 NormalizedFilePath,
+                                                 SMethod (..),
+                                                 TextDocumentIdentifier (TextDocumentIdentifier),
+                                                 TextEdit (TextEdit), Uri,
+                                                 WorkspaceEdit (WorkspaceEdit),
+                                                 type (|?) (InR))
+import           Text.Regex.TDFA                (AllTextMatches (getAllTextMatches),
+                                                 (=~))
+
+descriptor :: PluginId -> PluginDescriptor IdeState
+descriptor plId = (defaultPluginDescriptor plId) { pluginHandlers = mkPluginHandler STextDocumentCodeAction codeActionHandler }
+
+codeActionHandler :: PluginMethodHandler IdeState TextDocumentCodeAction
+codeActionHandler ideState plId CodeActionParams {_textDocument = TextDocumentIdentifier uri, _context = CodeActionContext (List diags) _} = response $ do
+      nfp <- getNormalizedFilePath plId (TextDocumentIdentifier uri)
+      decls <- getDecls ideState nfp
+      let actions = mapMaybe (generateAction uri decls) diags
+      pure $ List actions
+
+getDecls :: MonadIO m => IdeState -> NormalizedFilePath -> ExceptT String m [LHsDecl GhcPs]
+getDecls state = handleMaybeM "Error: Could not get Parsed Module"
+    . liftIO
+    . fmap (fmap (hsmodDecls . unLoc . pm_parsed_source))
+    . runAction "changeSignature.GetParsedModule" state
+    . use GetParsedModule
+
+-- | Text representing a Declaration's Name
+type DeclName = Text
+-- | The signature provided by GHC Error Message (Expected type)
+type ExpectedSig = Text
+-- | The signature provided by GHC Error Message (Actual type)
+type ActualSig = Text
+-- | The signature defined by the user
+type DefinedSig = Text
+
+-- | DataType that encodes the necessary information for changing a type signature
+data ChangeSignature = ChangeSignature {
+                         -- | The expected type based on Signature
+                         expectedType :: ExpectedSig
+                                       ,
```

Comment:
I've never seen this style of comma placement before! Up to you, but I think either before or after the field on the same line is more typical.

Reply:
Honestly, I just noticed it myself. I think I used a code action to generate comments and that's how it ended up

Reply:
Maybe a bug report for the haddock comment insertion plugin!